web/src/components/vis/PcaPage/PcaPage.vue

```diff
@@ -36,25 +39,45 @@ export default {
   },
 
   computed: {
-    pcX() {
-      return Number.parseInt(this.pcXval, 10);
-    },
-
-    pcY() {
-      return Number.parseInt(this.pcYval, 10);
-    },
-
-    numComponents() {
-      return Number.parseInt(this.numComponentsVal, 10);
-    },
-
     ready() {
       const pcaReady = this.$store.getters.ready(this.id, 'pca');
       const loadingsReady = this.$store.getters.ready(this.id, 'loadings');
 
       return pcaReady && loadingsReady;
     },
   },
+  watch: {
+    pcXval: {
+      handler(val) {
+        const pcX = Number.parseInt(val, 10);
+        // eslint-disable-next-line no-restricted-globals
+        if (!isNaN(pcX)) {
+          this.pcX = pcX;
+        }
+      },
+      immediate: true,
+    },
```

Comment:
This pattern feels bad, but I couldn't think of anything better for ensuring that when the input is set to the empty string, the value does not change.